p22/src/fibonacci.rs

```diff
@@ -0,0 +1,146 @@
+//! Fibonacci series - Recursive implementation
+//! # Examples
+//! ```
+//! use p22::fibonacci::fibonacci_recursive as fib;
+//! assert_eq!(0, fib(0));
+//! assert_eq!(1, fib(1));
+//! assert_eq!(1, fib(2));
+//! ```
+pub fn fibonacci_recursive(n: u32) -> u64 {
+    match n {
+        0 => 0,
+        1 => 1,
+        _ => fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2),
+    }
+}
+
+/// Fibonacci series - Non Recursive implementation
+/// # Examples
+/// ```
+/// use p22::fibonacci::fibonacci_non_recursive as fib;
+/// assert_eq!(0, fib(0));
+/// assert_eq!(1, fib(1));
+/// assert_eq!(1, fib(2));
+/// ```
+pub fn fibonacci_non_recursive(n: u32) -> u64 {
+    match n {
+        0 => return 0,
+        1 => return 1,
+        2 => return 1,
+        _ => None::<String>,
+    };
+    assert!(n >= 2);
+    let mut past_2 = fibonacci_non_recursive(0);
+    let mut past = fibonacci_non_recursive(1);
+    let mut number = fibonacci_non_recursive(2);
```

Comment:
You don't need to use `fibonacci_non_recursive` to initialize these constants.

Reply:
done

Reply:
You still use `fibonacci_non_recursive` in the latest commit. Have you forgot to commit changes in this file?